### ISSUE TITLE
Void Linux Building/Packaging

### ISSUE THREAD

**Issue by hwspeedy** [hardinfo2/hardinfo2#49]
Void Linux Building - **UPDATED TO 2.1.14**
----------
xbps-install git cmake gcc gettext curl make pkg-config
xbps-install json-glib-devel zlib-devel libsoup3-devel gtk+3-devel qt5-declarative-devel
git clone https://github.com/hardinfo2/hardinfo2
cd hardinfo2
./tools/git_latest_release.sh (Switch to latest stable release, tools/git_unstable_master.sh for developers)
mkdir build
cd build
cmake ..
make install -j   (Will pollute your distro!! - Please package...)
xbps-install lm_sensors sysbench lsscsi dmidecode udisks2 xdg-utils iperf3 fwupd glxinfo
hardinfo2

![image](https://github.com/hardinfo2/hardinfo2/assets/17836636/20c8c451-0fa1-4195-af89-1af7a78a3211)

**If anyone would like to package hardinfo2 for Void Linux - Just give a ping here, thanx**

**Comment by hwspeedy** [comment]
Hi @zlice @classabbyamp

hardinfo has been unmaintained for long time and Void Linux has some very old version from before Berlios shut down. Last 15 years it was here https://github.com/lpereira/hardinfo - and now transferred here to community edition as written in lpereira's README.md.

Would any of you be interested in updating void Linux with hardinfo2 package instead of old unmaintained hardinfo?

**Comment by hwspeedy** [comment]
@sgn
Would you consider updating hardinfo to hardinfo2 (community edition) for Void Linux?

If you don't have the time - maybe you know someone, who would - please add the person, thanx.

https://github.com/hardinfo2/hardinfo2/issues/49

**Comment by hwspeedy** [comment]
@tranzystorekk
Would you consider updating hardinfo to hardinfo2 (community edition) for Void Linux?

If you don't have the time - maybe you know someone, who would - please add the person, thanx.

https://github.com/hardinfo2/hardinfo2/issues/49

**Comment by tranzystorekk** [comment]
please don't mass ping void maintainers with package bump requests, either discuss it on irc or the github discussions for the void-packages repo

**Comment by hwspeedy** [comment]
Okay, sorry for the pings, thanx for responding.

zlice - if you want to package/build for libsoup2.4 instead of libsoup3 use `cmake -DHARDINFO2_LIBSOUP3=0` when building hardinfo2.

**Comment by hwspeedy** [comment]
Thanx zlice,

Is it possible to put in the package recommends for void packaging?

xbps-install lm_sensors sysbench lsscsi dmidecode udisks2 xdg-utils iperf3 fwupd glxinfo

**Comment by hwspeedy** [comment]
in most distros you can via the package info make it install other packages needed to make full use of the program. Hardinfo2 uses other packages to gain information and do benchmarking.

**From README.md:**
```
Package installs these:

sysbench: is needed to run standard sysbench benchmarks.
RiscV: The package is not ready yet - build by using the ./tools/build_and_install_sysbench_riscv.sh script from building hardinfo2.
udisks2: is needed to provide NVME++ informations.
dmi-decode: is needed to provide DMI informations.
mesa-utils: is needed to provide opengl and run standard sysbench benchmarks.
lsscsi: gives information about hard drives.
lm-sensors: is needed to provide sensors values.
xdg-utils: xdg_open is used to open your browser for bugs, homepage & links.
iperf3: iperf3 is used to benchmark internal network speed.
fwupd: fwupd is used to read and display information about firmware in system.
Service: Service loads SPD modules (at24/ee1004/spd5118) to display SPD info for your DIMMs memory. Show addresses for iomem+ioports.
xrandr/x11-xserver-utils: xrandr is used to read monitor setup
vulkan-tools: vulkaninfo is used to display vulcan information.
```

**Comment by zlice** [comment]
that's the `depends` line or probably better a separate 'meta' package like https://github.com/void-linux/void-packages/blob/master/srcpkgs/intel-video-accel/template

**Comment by hwspeedy** [comment]
Yes, depends are missing and maybe the service that is installed with the package. 
Service should be enabled and started on package install as most users will start using the program immediately. Read more about service here: https://github.com/hardinfo2/hardinfo2/issues/72, I use hardinfo2/tools/postinst for rpm/deb CPack packaging.

**Comment by ahesford** [comment]
Thanks for the heads up that hardinfo is severely outdated. https://github.com/void-linux/void-packages/pull/52277

**Comment by hwspeedy** [comment]
Regarding hardinfo2 service - I just discovered **runit in Void Linux** - first distro with this nice service system.

So a first shot for support for runit: https://github.com/hardinfo2/hardinfo2/commit/1538821acb4a87ee97aaf6676e8c4660df6ccfe9
1. Add hardinfo2/tools/run 
2. Include enabled and start in hardinfo2/tools/postinst
3. CMakelist.txt support for runit for Void Linux.

The package install process needs to run hardinfo2/tools/postinst or enable and start the one-shot service in some other way.

Works with make install and /etc/sv/hardinfo2/run or sv restart hardinfo2.

**Comment by ahesford** [comment]
We generally prefer to ship our own services, because most upstreams are not regular users of runit and we want to make sure that the services installed follow best practices for Void.

Also, unless a regular Void contributor accepts maintenance responsibility for a `hardinfo2`, it is unlikely to be considered for inclusion in the repository. We do not typically consider new package PRs submitted by upstream authors unless those authors are known, reliable Void Linux users and contributors. In any case, upstream promotion of projects via inclusion in the Void repository runs contrary to our bottom-up, demand-motivated consideration of new packages.

**Comment by hwspeedy** [comment]
That would be perfect, thanx.

Yes, someone using Void Linux and hardinfo2 should be the package maintainer for Void Linux.
I hope zlice would like to continue.

Thanx for all your help and let me know if you need any changes to hardinfo2, thanx.

**Comment by sgn** [comment]
> libsoup needs updating to 3.0+ `version=2.74.3`, current version is 3.6.0 which builds with adding nghttp2-devel

Use `libsoup3`
